### DISCUSSION TITLE
Status of database_grant versus snowflake_grant_privileges_to_role?

### DISCUSSION THREAD

**Discussion by chrisweis** [Snowflake-Labs/terraform-provider-snowflake#2433] [Q&A] [answered]
Hi,

I'm unclear on whether we should currently be trying to use `snowflake_grant_privileges_to_role` or `database_grant` to grant permissions at the database level?  Here is some example code that doesn't trigger a deprecation warning, and I'm wondering if I can migrate it or not?:

```
resource "snowflake_database_grant" "TST_DW_DB__ALL" {
  database_name = snowflake_database.TST_DW_DB.name

  privilege = "ALL PRIVILEGES"
  roles     = ["TERRAFORM_ROLE","DBT_TST_ROLE"]

  with_grant_option = false
}
```

Thanks,
Chris

**Comment by sfc-gh-jcieslak** [comment, marked as answer]
Hey @chrisweis 
So, somehow it's missing a deprecation message. I will add it, but yeah, `snowflake_grant_privileges_to_role` or even `snowflake_grant_privileges_to_account_role` (a newer version of `snowflake_grant_privileges_to_role`) should be a way to go.
